File: app/src/main/java/knf/kuma/changelog/ReleaseAdapter.java
package knf.kuma.changelog;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import knf.kuma.R;
import knf.kuma.changelog.objects.Changelog;
import knf.kuma.changelog.objects.Release;

/**
 * Created by Jordy on 08/02/2018.
 */

public class ReleaseAdapter extends RecyclerView.Adapter<ReleaseAdapter.ReleaseItem> {

    private List<Release> list = new ArrayList<>();

    public ReleaseAdapter(Changelog changelog) {
        this.list = changelog.releases;
    }

    @Override
    public ReleaseItem onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ReleaseItem(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_changelog, parent, false));
    }

    @Override
    public void onBindViewHolder(ReleaseItem holder, int position) {
        Release release = list.get(position);
        holder.version.setText(release.version);
        holder.code.setText(release.code);
        holder.recyclerView.setAdapter(new ChangeAdapter(release));
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class ReleaseItem extends RecyclerView.ViewHolder {
        @BindView(R.id.version)
        TextView version;
        @BindView(R.id.code)
        TextView code;
        @BindView(R.id.recycler)
        RecyclerView recyclerView;

        ReleaseItem(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/knf/kuma/changelog/ReleaseAdapter.kt
package knf.kuma.changelog

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import knf.kuma.R
import knf.kuma.changelog.objects.Changelog
import org.jetbrains.anko.find

internal class ReleaseAdapter(changelog: Changelog) : RecyclerView.Adapter<ReleaseAdapter.ReleaseItem>() {

    private val list = changelog.releases

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ReleaseItem {
        return ReleaseItem(LayoutInflater.from(parent.context).inflate(R.layout.item_changelog, parent, false))
    }

    override fun onBindViewHolder(holder: ReleaseItem, position: Int) {
        val release = list[position]
        holder.version.text = release.version
        holder.code.text = release.code
        holder.recyclerView.adapter = ChangeAdapter(release)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    internal inner class ReleaseItem(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val version: TextView = itemView.find(R.id.version)
        val code: TextView = itemView.find(R.id.code)
        val recyclerView: RecyclerView = itemView.find(R.id.recycler)
    }
}
